app/src/main/java/com/github/se/polyfit/ui/flow/AddMealFlow.kt

```diff
@@ -43,4 +62,29 @@ fun AddMealFlow(
           navigateForward = { goForward() })
     }
   }
+  if (showCancelDialog) CancelAddMealDialog(closeDialog = { showCancelDialog = false }, goBack)
+}
+
+@Composable
+private fun CancelAddMealDialog(closeDialog: () -> Unit, goBack: () -> Unit) {
+  val context = LocalContext.current
+
+  Dialog(onDismissRequest = closeDialog) {
+    GradientBox {
+      Column(
+          modifier = Modifier.fillMaxWidth().padding(16.dp),
+          horizontalAlignment = Alignment.CenterHorizontally) {
+            Text(context.getString(R.string.confirmAddMealCancel))
+            PrimaryButton(
+                text = context.getString(R.string.confirmDiscard),
+                onClick = goBack,
+                modifier = Modifier.testTag("GoBack"))
+            PrimaryButton(
+                text = context.getString(R.string.denyRequest),
+                color = PrimaryPink,
+                onClick = closeDialog,
+                modifier = Modifier.testTag("DenyButton"))
+          }
+    }
+  }
```

Comment:
As this screen is only relevant to the `IngredientScreen`, it might be a good idea to move it to that file if possible.